mlflow/pytorch/torchserve/deploy/serve.py

```diff
@@ -0,0 +1,388 @@
+        return deployment_list
+
+    def get_deployment(self, name):
+        """
+        Print the detailed description of the deployment with the name given at --name
+        in the specified target
+        """
+
+        url = "{}/{}/{}".format(self.management_api, "models", name)
+        resp = requests.get(url)
+        if resp.status_code != 200:
+            raise Exception(
+                "Unable to get deployments with name %s. Server returned status code %s and response: %s"
+                % (name, resp.status_code, resp.content)
+            )
+        return {"deploy": resp.text}
+
+    def predict(self, deployment_name, df):
+        """
+        Predict using the inference api
+        Takes dataframe or json string as input and returns output as string
+        """
+        version = self.server_config["version"]
+        url = "{}/{}/{}/{}".format(
+            self.inference_api, "predictions", deployment_name, version
+        )
+        if isinstance(df, pd.DataFrame):
+            df = df.to_json(orient="records")[1:-1].replace("},{", "} {")
+        try:
+            data = json.loads(df)
+        except (TypeError, json.decoder.JSONDecodeError) as e:
+            raise Exception(
+                "Input data can either be dataframe or Json string: {}".format(e)
+            )
+
+        resp = requests.post(url, data)
+        if resp.status_code != 200:
+            raise Exception(
+                "Unable to infer the results for the name %s. Server returned status code %s and response: %s"
+                % (deployment_name, resp.status_code, resp.content)
+            )
+
+        return resp.text
+
+    def __generate_mar_file(
+        self, model_name, version, model_file, handler_file, extra_files, model_uri
+    ):
+
+        """
+        Generates mar file using the torch archiver in the specified model store path
+        """
+        valid_file_suffixes = [".pt", ".pth"]
+        requirements_file = "requirements.txt"
+        req_file_path = None
+
+        if not os.path.isfile(model_uri):
+            path = Path(_download_artifact_from_uri(model_uri))
+            model_config = path / "MLmodel"
+            if not model_config.exists():
+                raise Exception(
+                    "Failed to find MLmodel configuration within the specified model's root directory."
+                )
+            else:
+                model_path = None
+                if path.suffix in valid_file_suffixes:
+                    model_uri = path
+                else:
+                    for file in path.iterdir():
+                        if file.is_dir():
+                            for sub_files in file.iterdir():
+                                if sub_files.suffix in valid_file_suffixes:
+                                    model_path = sub_files
+                                if PurePath(sub_files).name == requirements_file:
+                                    req_file_path = sub_files
```

Comment:
Just to confirm, what's the goal of this block of code / is this necessarily the structure of a `.mar` file? Should we use `os.path.walk` to recursively look through the contents of the archive, or just look for files in the same top-level directory as the model? In particular, the current code seems to assume that a requirements file is located either in the same top-level directory as the model, or one directory deep